prov/gni/configure.m4

```diff
@@ -39,6 +41,13 @@ AC_DEFUN([FI_GNI_CONFIGURE],[
                                  [alps_util_happy=0])
 	       ])
 
+	# Pull out gcc version number with regex backreference
+	gcc_version=`gcc --version | head -n1 | sed 's/.*\([[0-9]]\.[[0-9]]\.[[0-9]]\).*/\1/'`
```

Comment:
It would be a lot safer to use a test that looked like the code that GCC <4.9.0 fails to compile instead of trying to grep out the compiler version.  Also, what if the user's `$CC` isn't set to just `gcc` in the path but instead is `/path/to/gcc` or `prefixed-gcc` or `gcc-4.10.1`?  What about building it with `clang`?

Or even if you want to use a version number-based test, a more robust test would be to use `AC_COMPILE_IFELSE` and a test that compared `__GNUC__`/`__GNUC_MINOR__`/`__GNUC_PATCHLEVEL__`: https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html

What sort of support is present in GCC >=4.9.0 that the GNI provider requires?